agri-store-react/src/Components/ProductCard/ProductCard.js

```diff
@@ -0,0 +1,47 @@
+import React, { useContext } from "react";
+import { ProductListingPagecontext } from "../Context/ContextProductListingPage";
+import "./ProductCard.css";
+function ProductCard() {
+  const { productdata } = useContext(ProductListingPagecontext);
+
+  return (
+    <div className="productscard-in-productpage">
+      {productdata.map((pdata) => {
+        return (
+          <div class="horizontal-card">
+            <img
+              class="horizontal-card__image"
+              src={pdata.image}
+              alt="image_ecom"
+            />
+            <div class="horizontal-card__content">
+              <h2 class="horizontal-card__title">{pdata.title}</h2>
+              <h3 class="horizontal-card__price">Price : ${pdata.price}</h3>
+              <p class="horizontal-card__description"></p>
+              <h3 class="horizontal-card__ratings">Rating: {pdata.rating}</h3>
```

Comment:
Can destructure pdata in the beginning, so you won't have to prefix it everywhere.
Also I suggest using more descriptive name like productData, etc.